modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ITIgniteNodeRestartTest.java

```diff
@@ -0,0 +1,234 @@
+    }
+
+    /**
+     * Restarts a node with changing configuration.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void changeConfigurationOnStartTest() throws Exception {
+        Ignite ignite = IgnitionManager.start(NODE_NAME, null, workDir.resolve(NODE_NAME));
+
+        int nodePort = getNodePort(NODE_NAME);
+
+        assertEquals(47500, nodePort);
+
+        IgnitionManager.stop(ignite.name());
+
+        int newPort = 3322;
+
+        String updateCfg = "network.port=" + newPort;
+
+        ignite = IgnitionManager.start(NODE_NAME, updateCfg, workDir.resolve(NODE_NAME));
+
+        nodePort = getNodePort(NODE_NAME);
+
+        assertEquals(newPort, nodePort);
+
+        IgnitionManager.stop(ignite.name());
+    }
+
+    /**
+     * Checks that the only one non-default property overwrites after another configuration is passed on the node
+     * restart.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void twoCustomPropertiesTest() throws Exception {
+        String startCfg = "network: {\n" +
+            "  port:3344,\n" +
+            "  netClusterNodes:[ \"localhost:3344\" ]\n" +
+            "}";
+
+        Ignite ignite = IgnitionManager.start(NODE_NAME, startCfg, workDir.resolve(NODE_NAME));
+
+        assertEquals(3344, getNodePort(NODE_NAME));
+        assertArrayEquals(new String[] {"localhost:3344"}, getClusterNodes(NODE_NAME));
+
+        IgnitionManager.stop(ignite.name());
+
+        ignite = IgnitionManager.start(NODE_NAME, "network.netClusterNodes=[ \"localhost:3344\", \"localhost:3343\" ]", workDir.resolve(NODE_NAME));
+
+        assertEquals(3344, getNodePort(NODE_NAME));
+        assertArrayEquals(new String[] {"localhost:3344", "localhost:3343"}, getClusterNodes(NODE_NAME));
+
+        IgnitionManager.stop(ignite.name());
+    }
+
+    /**
+     * Gets current network port.
+     *
+     * @param nodeName Node name.
+     * @return Network port.
+     * @throws Exception If some issue has happened when getting the port.
+     */
+    private int getNodePort(String nodeName) throws Exception {
+        NetworkConfiguration networkCfgTree = getNetworkCfgTree(nodeName);
+
+        return networkCfgTree.port().value();
+    }
+
+    /**
+     * Gets an array of nodes.
+     *
+     * @param nodeName Node name.
+     * @return Array of node's hosts.
+     * @throws Exception If some issue has happened when getting the nodes.
+     */
+    private String[] getClusterNodes(String nodeName) throws Exception {
+        NetworkConfiguration networkCfgTree = getNetworkCfgTree(nodeName);
+
+        return networkCfgTree.netClusterNodes().value();
+    }
+
+    /**
+     * Gets a network configuration for a node with the name specified.
+     *
+     * @param nodeName Node name.
+     * @return Network configuration.
+     * @throws Exception If failed.
+     */
+    private NetworkConfiguration getNetworkCfgTree(String nodeName) throws Exception {
```

Comment:
I still don't like the usage of reflection here. I strongly disagree with this approach.

Reply:
Reworked with

public class IgniteImpl implements Ignite {
    /**
     * @return Node configuration.
     */
    public ConfigurationRegistry nodeConfiguration() {
        return nodeConfigurationMgr.configurationRegistry();
    }
}